scripts/gha-test

```diff
@@ -0,0 +1,172 @@
+# here for simplicity's sake, as it requires semver parsing & matching. The last release should
+# be good enough for our needs.
+export SOME_K8S_VERSION=$(curl -sS https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/dev-v2.8/data/data.json | jq -r ".$DIST.releases[-1].version")
+fi
+
+if [ -z "${CATTLE_CHART_DEFAULT_URL}" ]; then
+# If `CATTLE_CHART_DEFAULT_URL` is not set, use the `https://github.com/rancher/charts` so GitHub is used instead of
+# the default `https://git.rancher.io/charts` to reduce the reliance and load on our Git mirror
+export CATTLE_CHART_DEFAULT_URL=https://github.com/rancher/charts
+fi
+
+echo Starting rancher server for test
+touch /tmp/rancher.log
+
+export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+
+mkdir -p /var/lib/rancher/$DIST/agent/images
+grep PodTestImage ./tests/v2prov/defaults/defaults.go | cut -f2 -d'"' > /var/lib/rancher/$DIST/agent/images/pull.txt
+grep MachineProvisionImage ./pkg/settings/setting.go | cut -f4 -d'"' >> /var/lib/rancher/$DIST/agent/images/pull.txt
+mkdir -p /usr/share/rancher/ui/assets
+curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-amd64 -o /usr/share/rancher/ui/assets/rancher-system-agent-amd64
+curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-arm64 -o /usr/share/rancher/ui/assets/rancher-system-agent-arm64
+curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/system-agent-uninstall.sh -o /usr/share/rancher/ui/assets/system-agent-uninstall.sh
+
+run_rancher()
+{
+    RESTART_COUNT=0
+    while sleep 2; do
+        if [ "$PID" != "-1" ] && [ ! -e /proc/$PID ]; then
+            echo Rancher died
+            dump_rancher_logs
+            echo K3s logs were:
+            echo -e "-----K3S-LOG-DUMP-START-----"
+            cat build/testdata/k3s.log
+            echo -e "\n-----K3S-LOG-DUMP-END-----"
+            set +e
+            echo Attempting to kill K3s
+            pkill -e k3s
+            set -e
+            PID=-1
+            if [ "$RESTART_COUNT" = "2" ]; then
+                echo Rancher died 3 times, aborting
+                kill -42 $PWRAPPROC
+            fi
+            RESTART_COUNT=$((RESTART_COUNT + 1))
+            sleep 5
+        fi
+        if [ "$PID" = "-1" ]; then
+          echo Starting rancher server using run
+          ./scripts/gha-run >/tmp/rancher.log 2>&1 &
+          PID=$!
+        fi
+        sleep 2
+    done
+}
+
+dump_rancher_logs()
+{
+  echo Rancher logs were
+  echo -e "-----RANCHER-LOG-DUMP-START-----"
+  cat /tmp/rancher.log
+  echo -e "\n-----RANCHER-LOG-DUMP-END-----"
+}
+
+# uncomment to get startup logs. Don't leave them on because it slows drone down too
+# much
+#tail -F /tmp/rancher.log &
+#TPID=$!
+
+trap "exit 1" 42
+PWRAPPROC="$$"
+
+PID=-1
+run_rancher &
+RANCHER_RUN_PID=$!
+
```

Comment:
add this line

Reply:
https://github.com/rancher/rancher/blob/b573c957ab78a1121888c6b7d73d787efc479d7f/scripts/test#L141

Reply:
Added it.